airbyte-cdk/bulk/core/load/src/test/kotlin/io/airbyte/cdk/load/state/StreamManagerTest.kt

```diff
@@ -154,6 +154,18 @@ class StreamManagerTest {
                             Pair(stream1, ExpectComplete(true)),
                         )
                     ),
+                    TestCase(
```

Comment:
The code changes look correct, but the test scenarios aren't catching what we're after.

The scenario we want is more like:
* `AddPersisted(0, 4)`
* `AddComplete(5, 9)`
* `ExpectPersistedUntil(10, true)`

Ie, if one batch skips persisted and goes straight to complete, we still consider the results *persisted* (and emit checkpoint messages back to the platform)

Reply:
I think those lines would have passed with the old code, because of the check for `COMPLETE || PERSISTED` over the homogenous range.

Your change makes `COMPLETE => PERSISTED` even for heterogenous ranges.

Reply:
Yeap, though they'd fail in the old code because my change also changes `areRecordsPersistedUntil` to only look at PERSISTED.

Reply:
adding your case now!